sdk/keyvault/Azure.Security.KeyVault.Keys/tests/LocalCryptographyClientTests.cs

```diff
@@ -0,0 +1,394 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.Security.KeyVault.Keys.Cryptography;
+using NUnit.Framework;
+
+namespace Azure.Security.KeyVault.Keys.Tests
+{
+    public class LocalCryptographyClientTests : ClientTestBase
+    {
+        public LocalCryptographyClientTests(bool isAsync) : base(isAsync)
+        {
+            TestDiagnostics = false;
+        }
+
+        private byte[] TestData { get; } = Encoding.UTF8.GetBytes("test");
+
+        private byte[] TestKey { get; } = new byte[] { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07 };
+
+        private Stream TestStream => new MemoryStream(TestData);
+
+        [Test]
+        [SyncOnly]
+        public void LocalCryptographyClientRequiresJsonWebKey()
+        {
+            ArgumentException ex = Assert.Throws<ArgumentNullException>(() => new LocalCryptographyClient(null));
+            Assert.AreEqual("jsonWebKey", ex.ParamName);
+        }
+
+        [Test]
+        [SyncOnly]
+        public void LocalCryptographyClientRequiredJsonWebKeyType()
+        {
+            JsonWebKey jwk = new JsonWebKey(null);
+
+            Assert.Throws<NotSupportedException>(() => new LocalCryptographyClient(jwk));
+        }
+
+        [Test]
+        [SyncOnly]
+        public void KeyIdFromJsonWebKey()
+        {
+            JsonWebKey jwk = new JsonWebKey(null)
+            {
+                Id = nameof(KeyIdFromJsonWebKey),
+                KeyType = KeyType.Rsa,
+            };
+
+            LocalCryptographyClient client = new LocalCryptographyClient(jwk);
+
+            Assert.AreEqual(nameof(KeyIdFromJsonWebKey), client.KeyId);
+        }
+
+        #region Encrypt / Decrypt
+        [Test]
+        public async Task EncryptOperationNotSupported()
+        {
+            JsonWebKey jwk = new JsonWebKey(RSA.Create(), keyOps: Array.Empty<KeyOperation>());
+            LocalCryptographyClient client = CreateClient<LocalCryptographyClient>(jwk);
+
+            await this.CatchAsync<NotSupportedException>(async () => await client.EncryptAsync(new EncryptionAlgorithm("ignored"), TestData));
```

Comment:
`Assert.ThrowsAsync`?

Reply:
NUnit's doesn't correctly handle sync/async code like we use in our test framework/code. I'd have to pepper every test with a conditional and either `Assert.ThrowsAsync` or `Assert.Throws`, which is tedious compared to this (similar to what we use in Search as well).

Reply:
Hm, an instrumented client should work with `Assert.ThrowsAsync`. Can you expand on what doesn't work?

Reply:
I see it being used for other libraries too: https://github.com/Azure/azure-sdk-for-net/blob/3b69fc26c3ccf5ffd3f0a70100080a957d6e6488/sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormTrainingClient/FormTrainingClientLiveTests.cs#L126

Reply:
It wasn't working before - but I can't seem to replicate it. I dont think I was using `async/await` in the delegate (it's not typically needed given the param type), but I just tried again and even that seemed to work. I ended up using `async/await`, though. IIRC, expected exceptions weren't being caught in sync invocations.